### PR TITLE
Fix the Remoting 3.10.x release flow

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] - Backport Code signing changes from 3.11
- [x] - Update annotation indexer from 1.4 to 1.12 in order to prevent failure when listing constructor annotations

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -67,6 +67,8 @@ THE SOFTWARE.
     <project.build.outputEncoding>UTF-8</project.build.outputEncoding>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <findbugs.failOnError>false</findbugs.failOnError>
+    <!-- TODO: Required to override by a version with certchain support (MJARSIGNER-53) on @oleg-nenashev's machine. Remove once it's in upstream -->
```
Comment (jglick) [inline pom.xml]: Where is “upstream”? Link to the PR where you are adding this.

File: pom.xml [hunk 2]
```diff
@@ -198,7 +200,15 @@ THE SOFTWARE.
       <version>1.7</version>
       <type>jar</type>
       <optional>true</optional><!-- no need to have this at runtime -->
+    </dependency> 
+    <dependency>
+      <groupId>org.jenkins-ci</groupId>
+      <artifactId>annotation-indexer</artifactId>
+      <version>1.12</version>
+      <type>jar</type>
```
Comment (jglick) [inline pom.xml]: gratuitous

File: pom.xml [hunk 3]
```diff
@@ -198,7 +200,15 @@ THE SOFTWARE.
       <version>1.7</version>
       <type>jar</type>
       <optional>true</optional><!-- no need to have this at runtime -->
+    </dependency> 
+    <dependency>
+      <groupId>org.jenkins-ci</groupId>
+      <artifactId>annotation-indexer</artifactId>
+      <version>1.12</version>
```
Comment (jglick) [inline pom.xml]: Does this match the version used by the Jenkins core version you are targeting?

File: pom.xml (hunk 1 above)
Comment (oleg-nenashev) [reply pom.xml]: Links are provided in the original pull-request: https://github.com/jenkinsci/remoting/pull/190

File: pom.xml (hunk 3 above)
Comment (oleg-nenashev) [reply pom.xml]: Yes, 2.73.2 uses this version

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Comment by oleg-nenashev** [comment]
Merging in order to spin the release

**Pull request merged**